test/e2e/create_enable_force_delete_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e_test
+
+import (
+	"context"
+	"time"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"github.com/gardener/gardener-extension-registry-cache/test/common"
+	e2e "github.com/gardener/gardener-extension-registry-cache/test/e2e"
+)
+
+var _ = Describe("Registry Cache Extension Tests", func() {
```

Comment:
I am with mixed feelings about the e2e test. Right now, in the ForceDelete func we call Delete (which is questionable to me, see my previous comment). I am wondering whether we need to test the force deletion with e2e test if its implementation stays that simple.
For example you cannot easily drop the `ForceDelete` func - the code won't compile anymore. If I see a potential regression that causes the test to fail, I rather this regression to come from the Shoot flows or from the extension library. And not from the ForceDeletion implementation of this extension.
I am not that much against adding this e2e test. However I was wondering in which cases this test will fail.

Reply:
With current implementation force delete is a little different from normal delete, Keeping the test will be better option IMO, but I can remove it also.

Reply:
Okay, let's then keep it. If after its addition the test brings more overhead (maintenance, unrelated flakes) than value (catching regressions) then it would be my first candidate for e2e test removal. 😅